src/ucp/api/ucp.h

```diff
@@ -1243,6 +1340,43 @@ ucs_status_t ucp_worker_signal(ucp_worker_h worker);
 
 
 /**
+ * @ingroup UCP_WORKER
+ * @brief Accept connections on a local address of the worker object.
+ *
+ * This routine binds the worker object to a sock_addr which is set by the user.
+ * The worker will listen to incoming connection requests and upon receiving such
+ * a request from the remote peer, an endpoint to it will be created.
+ * The user's call-back will be invoked once the endpoint is created.
+ *
+ * @param [in]  worker           Worker object that is associated with the
+ *                               params object.
+ * @param [in]  params           User defined @ref ucp_worker_listener_params_t
+ *                               configurations for the @ref ucp_listener_h.
+ * @param [out] listener         A handler to the listener, to be used later for
+ *                               destroy.
+ *
+ * @return Error code as defined by @ref ucs_status_t
+ */
+ucs_status_t ucp_worker_listen(ucp_worker_h worker,
+                               const ucp_worker_listener_params_t params,
+                               ucp_listener_h listener);
+
+
+/**
+ * @ingroup UCP_WORKER
+ * @brief Stop accepting connections on a local address of the worker object.
+ *
+ * This routine unbinds the worker from the given handle and stops
+ * listening for incoming connection requests on it.
+ *
+ * @param [in] listener        A handle to the listener to stop listening on.
+ *
+ * @return Error code as defined by @ref ucs_status_t
+ */
+ucs_status_t ucp_worker_listen_destroy(ucp_listener_h listener);
```

Comment:
shouldn't we keep 'worker' in the name? so that it's clear that ucp_worker_listen and the destroy func go together and that destroy is done per worker?